crates/solver/src/driver_logger.rs

```diff
@@ -129,7 +130,7 @@ impl DriverLogger {
                 // of race conditions as hard errors.
                 tracing::warn!(settlement_id, ?err, "Failed to submit settlement",);
                 self.metrics
-                    .settlement_submitted(err.as_outcome(), solver_name);
+                    .settlement_submitted(err.as_outcome(), solver_name, err.strategy);
```

Comment:
Looks like this could take the new error + metadata type directly without passing its member fields as parameters.

Reply:
Irrelevant after latest changes.